### ISSUE TITLE
Bug: VM Interface vlan list in UI uses invalid option

### ISSUE THREAD

**Issue by tlourey** [nautobot/nautobot#5058]
### Environment
* Nautobot version (Docker tag too if applicable): 2.1.0
* Python version: 3.10
* Database platform, version: postgres, 14.10
* Middleware(s):

### Steps to Reproduce
1. Create a VLAN
2. Create a Cluster Type
3. Create a Cluster
4. Create a VM in the cluster
5. Create a interface on the vm
6. Selected Access, Tagged or Tagged (all)

### Expected Behavior
Show a list of vlans

### Observed Behavior
Dropdown shows "The results could not be loaded"

Was able to replicate on [demo.nautobot.com](https://demo.nautobot.com)

It appears that this dropdown tries to use an API query attribute/option called `location_id` however this seems to generate an 400. 

Looking at the API browser with `GET /api/ipam/vlans/?q=&limit=50&offset=0&depth=0&location_id=<<GUID>>` or even `GET /api/ipam/vlans/?q=&limit=50&offset=0&depth=0&location_id=null` returns 

```
HTTP 400 Bad Request
API-Version: 2.1
Allow: GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "location_id": [
        "Unknown filter field"
    ]
}
```

**Comment by glennmatthews** [comment]
Thanks for the report! Looks like there are quite a few cases in `nautobot/virtualization/forms.py` that are using `location_id` as a filter parameter, which should generally be `location` instead.

**Comment by tlourey** [comment]
Hi @glennmatthews, still very very new to the model, but is it just a matter of the references under query_params being changed from
```  
        query_params={
            "location_id": "null",
        },
``` 
to 
```  
        query_params={
            "location": "null",
        },
``` 
?

If so, I'll give it a crack and if it pass any local linting & tests, follow the contributing guide to submit a PR if it will help. If there's more to it, I'll keep out of the way :D

**Comment by anton-wg2** [comment]
Hello.
I still have the problem on 2.1.3
It seems the wrong filter is still there
```
[Tue Feb  6 09:35:49 2024] GET /api/ipam/vlans/?q=&limit=50&offset=0&depth=0&location=null&location_id=e8573f65-41cd-40cc-a192-70515809c689 => generated 187953 bytes in 33 msecs (HTTP/1.1 400) 9 headers in 447 bytes (1 switches on core 0)
```
so `location=null` is fine, but `location_id=<uuid>` fails

**Comment by glennmatthews** [comment]
Looks like the fix was incomplete - see also analysis at https://github.com/nautobot/nautobot/pull/5124#issuecomment-1907824331. @tlourey were you going to be able/willing to submit a followup PR, or would you need us to run with fixing this?

**Comment by tlourey** [comment]
My apologies for my delay. I'll try and get to this in the next few days.

**Comment by tlourey** [comment]
> Looks like the fix was incomplete - see also analysis at [#5124 (comment)](https://github.com/nautobot/nautobot/pull/5124#issuecomment-1907824331). @tlourey were you going to be able/willing to submit a followup PR, or would you need us to run with fixing this?

Draft PR #5286 running tests now
